src/libraries/CustomRevert.sol

```diff
@@ -75,23 +78,33 @@ library CustomRevert {
         }
     }
 
-    /// @notice bubble up the revert message returned by a call and revert with the selector provided
-    /// @dev this function should only be used with custom errors of the type `CustomError(address target, bytes revertReason)`
-    function bubbleUpAndRevertWith(bytes4 selector, address addr) internal pure {
+    /// @notice bubble up the revert message returned by a call and revert with a wrapped ERC-7751 error
+    function bubbleUpAndRevertWith(address target, bytes4 functionSelector, bytes4 additionalContext) internal pure {
```

Comment:
consider renaming these to be a bit more clear, maybe:
address revertingContract
address revertingFunctionSelector
bytes4 additionalContext

Reply:
also -- why is `additionalContext` only bytes4 here but bytes in the custom error? seems context could be a custom error w/ params so should we support that here too?

Reply:
I guess keeping it bytes4 makes the encoding more simple if we specifically dont plan on having params here

Reply:
any additional context we had in the reverts was the target address (hook, erc20 token, recipient) based on the action, so we only needed 4bytes for the selector